eventhubs/data-plane/azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/BatchOptions.java

```diff
@@ -47,17 +47,33 @@
      * The partitionKey to use for all {@link EventData}s sent in the current {@link EventDataBatch}.
      * Setting a PartitionKey will deliver the {@link EventData} to a specific Event Hubs partition.
      */
-    public String partitionKey = null;
+    private String partitionKey = null;
```

Comment:
We can't change the visibility of fields - this is breaking API. Add suppression instead.